dxr/plugins/descriptor/__init__.py

```diff
@@ -9,6 +9,16 @@
 import dxr.indexers
 
 
+def describe_readme(lines):
+    """Return a string that represents a description for the given lines of a
+    presumed readme file or None if it can extract no suitable description.
+    """
+    # For now the heuristic is just the first non-empty line.
+    for candidate in (line.strip() for line in lines):
```

Comment:
Or even `return first(ifilter(None, (line.strip() for line in lines)))`, using more_itertools.first.